c/src/nnstreamer-capi-util.c

```diff
@@ -261,6 +294,8 @@ ml_tensors_info_set_count (ml_tensors_info_h info, unsigned int count)
     return ML_ERROR_INVALID_PARAMETER;
 
   tensors_info = (ml_tensors_info_s *) info;
+
+  /* This is atomic. No need for locks */
```

Comment:
I am not sure if this is an atomic operation, as this assignment can be broken down into more than 1 instruction by the compiler.

Reply:
32bit (unsigned int) assignment cannot be more than 1 instruction. We may incur multiple instructions for redirections (```tensors_info->num_tensors```) and load (```LOAD Rx count```).
However, these does not affect the atomicity of assignment; they are allowed to be executed out of the critical section.
We cannot keep it serialized with destruction; thus, redirection is not protected anyway.
The location of ```count``` is never affected by the serialization problem; thus, atomicity is not affected.

Thus, we simply need to protect ```MOV Ra Rb``` instruction here.